src/modules/keyboardmanager/dll/dllmain.cpp

```diff
@@ -273,6 +273,12 @@ class KeyboardManager : public PowertoyModuleIface
     // Function called by the hook procedure to handle the events. This is the starting point function for remapping
     intptr_t HandleKeyboardHookEvent(LowlevelKeyboardEvent* data) noexcept
     {
+        // Check if there is a registered KeyDelay for this key.
+        if (keyboardManagerState.HandleKeyDelayEvent(data))
+        {
+            return 1;
+        }
+

```

Comment:
Is it possible to move this into the `if (keyboardManagerState.DetectSingleRemapKeyUIBackend(data))` and the `if (keyboardManagerState.DetectShortcutUIBackend(data))` checks below? The issue with keeping it outside is that in the current implementation the order of handling is

- Detect Single Key UI
- Single Key Remap
- Detect Shortcut UI
- Shortcut Remap

The reason for that order is that we want the single key remapping to reflect in the detect shortcut UI. So if I have already remapped `a` to `b`, and I press Ctrl+A on my keyboard in the Type shortcut window, it should show Ctrl+B. So for the user the single key remap appears as the highest priority which is already done before the shortcut remaps.
With the current change, if HandleKeyDelay is kept at the start, and I remap Enter to say Shift, then when I click Enter on the detect shortcut screen it still shows Enter instead of Shift. By moving it below, we shouldn't have to separately worry about remembering that Shift is now Enter and hence holding Shift for x seconds should also make the window exit.

Reply:
I think a much simpler solution to the wrong mapping in the Detect Shortcut issue is to just Register the KeyDelay object with the mapped key instead of VK_RETURN and VK_ESCAPE, using the SingleKeyReMap map. Those maps don't really change while the key delaying is happening, so I think that would work.

Reply:
That might become more complicated when we add in the registry remap approach, since in that case we will only be receiving messages for the remapped key. What currently happens if we put the HandleKeyDelay inside the DetectSingleRemapKeyUI condition? Is there a situation where that condition is false (i.e. the UI is not up) but we need to handle something?

Reply:
No, it was just that I thought it might be simpler the other way, but I was wrong. I moved the HandleKeyDelay like you suggested and it worked perfectly fine.

Reply:
It should be fixed now